In [1]:
import os
os.getcwd()

'C:\\Users\\volle\\OneDrive\\COLLEGE\\Senior\\Summer 23\\Mee 460\\Project 2'

In [2]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns


In [3]:
Data_df = pd.read_csv("Project2 New Data Set.csv")
Data_df

Unnamed: 0      time_stamp  AmbientConditions_AmbientHumidity_U_Actual  \
0               0  3/6/2019 10:52                                       17.24   
1               1  3/6/2019 10:52                                       17.24   
2               2  3/6/2019 10:52                                       17.24   
3               3  3/6/2019 10:52                                       17.24   
4               4  3/6/2019 10:52                                       17.24   
...           ...             ...                                         ...   
14083       14083  3/6/2019 14:47                                       13.84   
14084       14084  3/6/2019 14:47                                       13.84   
14085       14085  3/6/2019 14:47                                       13.84   
14086       14086  3/6/2019 14:47                                       13.84   
14087       14087  3/6/2019 14:47                                       13.84   

       AmbientConditions_AmbientTemperature_U_Actual  \
0                                              23.53   
1                                              23.53   
2                                              23.53   
3                                              23.53   
4                                              23.53   
...                                              ...   
14083                                          24.43   
14084                                          24.43   
14085                                          24.43   
14086                                          24.43   
14087                                          24.43   

       Machine1_RawMaterial_Property1  Machine1_RawMaterial_Property2  \
0                               11.54                             200   
1                               11.54                             200   
2                               11.54                             200   
3                               11.54                             200   
4                               11.54                             200   
...                               ...                             ...   
14083                           12.22                             201   
14084                           12.22                             201   
14085                           12.22                             201   
14086                           12.22                             201   
14087                           12.22                             201   

       Machine1_RawMaterial_Property3  Machine1_RawMaterial_Property4  \
0                              963.00                             247   
1                              963.00                             247   
2                              963.00                             247   
3                              963.00                             247   
4                              963.00                             247   
...                               ...                             ...   
14083                         1048.06                             252   
14084                         1048.06                             252   
14085                         1048.06                             252   
14086                         1048.06                             252   
14087                         1048.06                             252   

       Machine1_RawMaterialFeederParameter_U_Actual  \
0                                           1241.26   
1                                           1246.09   
2                                           1246.29   
3                                           1247.59   
4                                           1252.83   
...                                             ...   
14083                                       1263.76   
14084                                       1255.42   
14085                                       1257.52   
14086                                       1261.99   
14

In [5]:
Data_df.describe()

Unnamed: 0  AmbientConditions_AmbientHumidity_U_Actual  \
count  14088.000000                                14088.000000   
mean    7043.500000                                   15.330759   
std     4066.999631                                    1.188993   
min        0.000000                                   13.840000   
25%     3521.750000                                   14.040000   
50%     7043.500000                                   15.120000   
75%    10565.250000                                   16.630000   
max    14087.000000                                   17.240000   

       AmbientConditions_AmbientTemperature_U_Actual  \
count                                   14088.000000   
mean                                       23.843635   
std                                         0.373535   
min                                        23.020000   
25%                                        23.530000   
50%                                        23.930000   
75%                                        24.160000   
max                                        24.430000   

       Machine1_RawMaterial_Property1  Machine1_RawMaterial_Property2  \
count                    14088.000000                    14088.000000   
mean                        11.851256                      205.676320   
std                          0.510309                       11.606324   
min                         11.540000                      200.000000   
25%                         11.540000                      200.000000   
50%                         11.540000                      200.000000   
75%                         12.220000                      201.000000   
max                         12.900000                      236.000000   

       Machine1_RawMaterial_Property3  Machine1_RawMaterial_Property4  \
count                    14088.000000                    14088.000000   
mean                       951.679815                      248.868896   
std                        126.662010                        3.297820   
min                        601.110000                      247.000000   
25%                        963.000000                      247.000000   
50%                        963.000000                      247.000000   
75%                       1027.430000                      251.000000   
max                       1048.060000                      257.000000   

       Machine1_RawMaterialFeederParameter_U_Actual  \
count                                  14088.000000   
mean                                    1242.764277   
std                                       95.845947   
min                                      231.300000   
25%                                     1257.170000   
50%                                     1264.440000   
75%                                     1273.460000   
max                                     1331.820000   

       Machine1_Zone1Temperature_C_Actual  Machine1_Zone2Temperature_C_Actual  \
count                        14088.000000                        14088.000000   
mean                            72.013330                           72.013107   
std                              0.063183                            0.406312   
min                             71.900000                           71.300000   
25%                             72.000000                           71.600000   
50%                             72.000000                           72.000000   
75%                             72.000000                           72.400000   
max                             72.500000                           72.700000   

       ...        Error5        Error6        Error7        Error8  \
count  ...  14088.000000  14088.000000  14088.000000  14088.000000   
mean   ...      2.609997      2.922971      1.860152      1.461759   
std    ...      0.570465      1.074306      1.413938      4.621794   
min    ...      0.000000      0.000000      0.000000      0.000000   
25%    .

In [16]:
Data_df[["Error0","Error1","Error2","Error3","Error4","Error5","Error6","Error7","Error8","Error9","Error10","Error11","Error12","Error13","Error14"]].describe(include="all")

Error0        Error1        Error2        Error3        Error4  \
count  14088.000000  14088.000000  14088.000000  14088.000000  14088.000000   
mean       0.806048     14.599097      1.668367      0.474661      1.433262   
std        0.446368      6.947074      0.645052      1.622527      2.969192   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.700000      8.860000      1.510000      0.330000      0.830000   
50%        0.780000      9.475000      1.710000      0.370000      1.010000   
75%        0.860000     22.740000      1.820000      0.400000      1.630000   
max       13.750000     25.870000     17.950000     21.880000     40.240000   

             Error5        Error6        Error7        Error8        Error9  \
count  14088.000000  14088.000000  14088.000000  14088.000000  14088.000000   
mean       2.609997      2.922971      1.860152      1.461759      1.478841   
std        0.570465      1.074306      1.413938      4.621794      4.041120   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.740000      2.430000      0.070000      0.350000      0.460000   
50%        2.740000      2.610000      2.970000      0.400000      0.570000   
75%        2.740000      4.250000      2.970000      0.430000      0.680000   
max        3.319891      5.000000      3.730000     21.300000     19.520000   

            Error10       Error11       Error12       Error13       Error14  
count  14088.000000  14088.000000  14088.000000  14088.000000  14088.000000  
mean       0.937230      4.644815      0.810021      0.447543      7.714686  
std        0.969435      2.554145      0.660151      0.850455      7.400059  
min        0.000000      0.000000      0.000000      0.000000      0.000000  
25%        0.730000      0.730000      0.450000      0.100000      1.960000  
50%        0.810000      6.160000      0.520000      0.140000      2.680000  
75%        0.920000      6.160000      0.630000      0.230000     17.720000  
max        8.650000      6.160000      2.020000      4.390000     24.270000

In [1]:
pip install PyPDF2

     -------------------------------------- 232.6/232.6 kB 4.7 MB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.


In [2]:
import PyPDF2 as pdf

In [3]:
from PyPDF2 import PdfReader

In [6]:
file = open("Project_2.pdf","rb")
reader = PdfReader(file)

In [32]:
import os
os.getcwd()

'C:\\users\\volle\\OneDrive\\COLLEGE\\Senior\\Summer 23\\Mee 460\\Project 2\\Snips'

In [33]:
os.chdir("C:\\users\\volle\\OneDrive\\COLLEGE\\Senior\\Summer 23\\Mee 460\\Project 2")
os.getcwd()

'C:\\users\\volle\\OneDrive\\COLLEGE\\Senior\\Summer 23\\Mee 460\\Project 2'

> 








# Project 2 Study Draft Version

> 
 
<font size=4>
    
1. Determine variations within features (max, min, mean)

1. Are there highly correlated features?

1. Visual EAD. Create plots among features or features vs targets (pick one calculated error as a target). This information will help to present your findings.

1. List any insights you gained

1. Show all your results in Jupyter notebook

> 

## 1.a)
 
<font size=4>

Looking at the figure below that is listing all the control features, we can make these conclusions. 
    
* Majority of the controlled temperature measurements even so they have a good distribution exhibits very little change in value around 1.5% or less, other than  
"Machine 1 Exit Zone Temperature C Actual". This controlled feature may not have much impact on the error measurements.    

* From the control factors only the motor speed shows 17-20% variation between min and max with an even distribution  
in comparison to the rest of the data.  
    
* Machine 1 Zone 2 Temperature Actual has a good distribution of data, however this is over a very small temperature  
range of 71.3 to 72.7 units

* Machine 2 Motor RPM also has a good distribution of data, also over a small range of 13.82 RPM to 13.97 RPM. </font> 


<img src="j1.png">

> 
# _____________________________________________________________________________________________________________________________________________

## 1.b)
 
<font size=4>
    
The figure below show setpoint values. The setpoints are predominantly set to one value however  
some data points are zero. Since this data shows no trends this data alone are not meaningful for any correlation study, however they are used in error calculations. 

But these set values could be modified in the future as a function of some features. </font>


<img src="2.png">

>
# _____________________________________________________________________________________________________________________________________________

## 1.c)
 
<font size=4>
    
The figure below are the actual measurements. Based on the distribution we can conclude that whenever the minimum is zero and the distribution has a large  
bar on the left most side, the data has a significant amount of zeros as a measurement. This will significantly distort the statistical data as well as  
any strong correlation </font>

<img src="3.png">

>
# _____________________________________________________________________________________________________________________________________________

## 1.d)
 
<font size=4>

The figure below shows the distribution of the error values.  
    
Similar to the figures above, the figure below shows a significant portion values to be zero, what that means in this  
case is the setpoint and actual measurements are the same, or both are zeros. For all the bars on the right most side  
of the distribution have probability that only one of the two, either the actual or setpoint is zero therefore the  
error appears to be very large.

Error 6 has the least biased and the widest distribution from all errors therefore this error makes the most sense to  
use in this analysis to find a correlation with other variables so it could be corrected to minimize the error.



<img src="4.png">

>
# _____________________________________________________________________________________________________________________________________________

## 2.a)
 
<font size=4>
    
Observing the figure of high correlations below, there were 111 correlations at a level of +/-1.000. However these were correlations  
of different measurements at setpoint, setpoint measurements were set to certain value so there is a 100% correlation. The same  
situation occurs when you match trends with properties. These correlations will not be considered in our modeling since  
they are not controlled.

<img src="5.png">

>
# _____________________________________________________________________________________________________________________________________________

## 2.b)
 
<font size=4>
    
From the figure below there is an observable high correlation between errors and the associated actual measurement. When  
considering error uses setpoint in its calculation, which is an unchanging value, it makes sense when the actual  
measurement increases the will error increase too. Only errors will used while disregarding measurements as a result of the previous statement.      

<img src="6.png">

>
# _____________________________________________________________________________________________________________________________________________

## 2.c)
 
<font size=4>
    
Correlation study conclusion: 

* Based on a distribution study, error 6 was the most appropriate feature to use for studying the correlations.

* In the figure the highest correlated features with error 6 are shown below. Cause of this variation and the good correlation is when a part is measured at a  
higher temperature it will measure larger due to heat expansion of material. A suggestion to the manufacture when measuring a part  
to do so in a temperature controlled environment


<img src="7.png">

>
# _____________________________________________________________________________________________________________________________________________

## 3.a)
 
<font size=4>
    
This is a scatter plot showing the strong correlation between error 6 and the Machine 1 Material Temperature. The color of the  
data is with respect to the Machine 1 motor amperage.

<img src="8.png">

>
# _____________________________________________________________________________________________________________________________________________

## 3.b)
 
<font size=4>
    
The figure below, even though is not related to error but it's for feature 1 but is selected becuase it is a good example of a strong 3D correlation of Machine3  
Motor RPM actual versus Feature 1 with the color of the data based on Ambient Humidity.

<img src="9.png">

>
# _____________________________________________________________________________________________________________________________________________

## 3.c)
 
<font size=4>
    

A very strong correlation is shown between Machine 3 Motor RPM and the Ambient humidity. In this case since RPM is a  
controlled 
factor, it can be adjust in the model based on humidity.

<img src="10.png">

>
# _____________________________________________________________________________________________________________________________________________

## 4) List of Observations
 
<font size=4>
    
* We don't know what is happening to the data with respect to time since, time is not a numerical value that could be used, we  
    only know how many times a measurement occurs ex (tool is wearing over time)

* Majority of the controlled temperature measurements experienced very little change in value around 1.5% or less, other than  
"Machine 1 Exit Zone Temperature C Actual".
This controlled feature may not have much impact on the error measurements.    

* From the control factors only the motor speeds there is 17-20% variation between min and max with an even distribution  
in comparison to the rest of the data. 
    
* Motor speed control is the factor that has the most impact on our target error 6 therefor could be used in our model as a  
controlling factor 
    
* All setpoint values are set to a specific number but few still have zero as a value, this will distort the min and the mean values.
    
* Whenever the minimum is zero and the distribution has a large bar on the left most side, the data has a significant amount of  
zeros as a measurement.
    
* Error 6 as a target was used for a correlation study since it was the least biased and the widest distribution.
    